torch/special/__init__.py

```diff
@@ -505,6 +505,32 @@
 Alias for :func:`torch.round`.
 """)
 
+log_softmax = _add_docstr(_special.special_log_softmax,
+                          r"""
+Computes softmax followed by a logarithm.
+
+While mathematically equivalent to log(softmax(x)), doing these two
+operations separately is slower, and numerically unstable. This function
+uses an alternative formulation to compute the output and gradient correctly.
+
+.. math::
+    \text{log\_softmax}(x_{i}) = \log\left(\frac{\exp(x_i) }{ \sum_j \exp(x_j)} \right)
+""" + r"""
+
+Args:
+    input (Tensor): input
+    dim (int): A dimension along which log_softmax will be computed.
```

Comment:
What happens when dim is None vs. what happens when a dimension is provided?

Reply:
`dim` is a mandatory argument so can't be None.